### PR TITLE
Hide version in footer for unauthenticated users

### PR DESCRIPTION
**Pull request by Jonher937**
Closes: #2682

# What's Changed
- Nautbot version number is hidden from the footer for unauthenticated users

# TODO
- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example

> Note the version number being present next to the hostname in the `before`, but not in the `after`.

## Before
before

## After
after

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
This looks reasonable enough, although as @Kircheneer pointed out, there's still the unauthenticated `/api/status/` endpoint to consider from a security standpoint (#1610).

Would you be willing to add a unit test for this rendering logic, probably in `nautobot.core.tests.test_views.HomeViewTestCase`?

**Comment by Jonher937** [comment]
> This looks reasonable enough, although as @Kircheneer pointed out, there's still the unauthenticated `/api/status/` endpoint to consider from a security standpoint (#1610).
> 
> Would you be willing to add a unit test for this rendering logic, probably in `nautobot.core.tests.test_views.HomeViewTestCase`?

That was not my experience, see my comment in the issue.

Sure, the devcontainer seems broken for VSCode (requires "db" but db is not defined, amongst other things). I've pushed a commit with a test, let's hope I nailed it, but noone usually does on the first try :sweat_smile:

**Review by glennmatthews — APPROVED** [review]
Looks good to me. We might want to revisit the test case at some point to make it more robust but it's definitely good enough for now. Thank you!

**Comment by Jonher937** [comment]
Sorry for the re-reviews. The CI only runs partial jobs since I'm a new contributor.

File: nautobot/core/tests/test_views.py [hunk 1]
```diff
@@ -79,6 +79,16 @@ def test_search_bar_visible_if_hide_restricted_ui_False(self):
...
 
+    def test_footer_version_visible_authenticated_users_only(self):
+        url = reverse("home")
+        response = self.client.get(url)
+        response_content = response.content.decode(response.charset).replace("\n", "")
+
+        footer_hostname_version_pattern = re.compile('<p class="text-muted">\\s+\\S+\\s+\\(v[\\d.]+\\)\\s+<\\/p>')
+        self.assertRegex(response_content, footer_hostname_version_pattern)
+        self.client.logout()
+        self.assertNotRegex(response_content, footer_hostname_version_pattern)
```
Comment (glennmatthews) [inline nautobot/core/tests/test_views.py:90]:
```suggestion
        self.client.logout()

        response = self.client.get(url)
        response_content = response.content.decode(response.charset).replace("\n", "")
        self.assertNotRegex(response_content, footer_hostname_version_pattern)
```

**Pull request merged**